src/heliaServer.ts

```diff
@@ -71,6 +91,9 @@ export class HeliaServer {
         path: '/',
         type: 'GET',
         handler: async (request, reply): Promise<void> => {
+          if (USE_SUBDOMAINS && request.hostname.split('.').length > 1) {
+            return this.fetch({ request, reply })
+          }
```

Comment:
if subdomains are enabled, and we're requesting the root of that subdomain, then call fetch.